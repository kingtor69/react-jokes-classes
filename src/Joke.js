import React from "react";
import "./Joke.css";

class Joke extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      vote: this.props.vote,
      votes: this.props.votes
    }
    this.upVote = this.upVote.bind(this);
    this.downVote = this.downVote.bind(this);
  };

  upVote() {
    this.setState({vote: this.state.vote(this.props.id, +1)});
  };

  downVote() {
    this.setState({vote: this.state.vote(this.props.id, -1)});
  };

  render() {
    const { text, id } = this.props;
    return (
      <div className="Joke">
        <div className="Joke-votearea">
          <button onClick={this.upVote}>
            <i className="fas fa-thumbs-up" />
          </button>

          <button onClick={this.downVote}>
            <i className="fas fa-thumbs-down" />
          </button>

          {this.state.votes}
        </div>

        <div className="Joke-text">{text}</div>
      </div>
    );
  };
};

export default Joke;
