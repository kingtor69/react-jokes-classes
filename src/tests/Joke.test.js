import React from 'react';
import { render } from '@testing-library/react';
import Joke from '../Joke';
import { Joke as StarterJoke } from '../starter-src/Joke';

it('should render without crashing', () => {
  render(Joke);
});

it('should render snapshot', () => {
  const { asFragment } = render();
  expect(asFragment()).toMatchSnapshot();
});

it('should match starter code snapshot', () => {
    const { asFragment } = render(StarterJoke);
    expect(asFragment()).toMatchSnapshot();
});
