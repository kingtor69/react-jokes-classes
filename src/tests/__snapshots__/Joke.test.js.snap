// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match starter code snapshot 1`] = `<DocumentFragment />`;

exports[`should render snapshot 1`] = `<DocumentFragment />`;
